spec/controllers/repositories_controller_spec.rb

```diff
@@ -12,9 +16,8 @@
   describe "GET #index" do
 
     it "assigns all repositories as @repositories" do
-      repository = create(:repository)
       get :index, {}, valid_session
-      expect(assigns(:repositories)).to eq([repository])
+      expect(assigns(:repositories)).to eq([visible_repository])
```

Comment:
There's a hidden problem here: `let` is lazy evaluated, meaning `invisible_repository` might not even exist at this point in time. Please switch to `let!`.

[This](https://www.relishapp.com/rspec/rspec-core/v/2-6/docs/helper-methods/let-and-let) section of Rspec's docs explains the behaviour pretty well